src/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { ReactNode, useState } from 'react';
+import Button from '@mui/material/Button';
+import ConfirmationDialog from './Confirmation/ConfirmationDialog';
+
+export type DialogProps = {
+  children: ReactNode;
+  onCancel: () => void;
+  onConfirm?: () => void;
+  sx?: { [key: string]: string | number };
+  onSubmit: () => void;
+};
+
+const Dialog: React.FC<DialogProps> = ({ onSubmit, onCancel, onConfirm, children, sx }) => {
+  const [isOpen, setIsOpen] = useState(false); // State to manage the open status of the confirmation dialog
+  
+  const handleOpen = () => setIsOpen(true);
+  const handleClose = () => setIsOpen(false);
+  
+  const handleConfirm = () => {
+    if (onConfirm) {
+      onConfirm();
+    }
+    handleClose(); // Close the confirmation dialog after confirming
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    if (onConfirm) {
+      handleOpen(); // Open the confirmation dialog
+    } else {
+      onSubmit(); // If no confirmation needed, proceed with submit
+    }
+  };
+
+  return (
+    <div style={sx}>
+      <form onSubmit={handleSubmit}>
+        {children}
+        <Button type="button" onClick={onCancel}>Cancel</Button>
+        <Button type="submit">Submit</Button>

```

Comment:
Conditionally render - only show the submit/confirm and cancel buttons if the `onConfirm` function is defined. Otherwise, if only `onCancel` is defined, we just show the `X` in the top right - no buttons.